Lib/statistics.py

```diff
@@ -304,6 +304,19 @@ def _fail_neg(values, errmsg='negative value'):
             raise StatisticsError(errmsg)
         yield x
 
+def _isqrt_frac_rto(n: int, m: int) -> float:
+    'Square root of n/m, rounded to the nearest integer using round-to-odd.'
+    a = math.isqrt(n*m) // m
+    return a | (a*a*m != n)
+
+def _sqrt_frac(n: int, m: int) -> float:
+    'Square root of n/m as a float, correctly rounded.'
+    q: int = (n.bit_length() - m.bit_length() - 109) // 2
```

Comment:
Maybe we should consider replacing `109` with `(3 + 2 * sys.float_info.mant_dig)`? That makes it look a bit less like a magic number, and in theory would give us _some_ protection in the unlikely event that this code ever meets a non-IEEE 754 floating-point system. (I have to say that I find it hard to care much about supporting non-IEEE 754 systems until we have  evidence that those systems exist and people want to run Python on them.)

Reply:
Added a comment so the 109 will be less magical.

How well would *_sqrt_frac()* degrade in an environment without other than 53 precision?

If it doesn't degrade nicely, I could add a fallback to *math.sqrt()*:

    if sys.float_info.mant_dig != 53:
        _sqrt_frac = math.sqrt